IHP/Job/Dashboard.hs

```diff
@@ -0,0 +1,530 @@
+    beforeAction = authenticate @authType
+    action AllJobsAction = autoRefresh $ indexPage @(jobs)
+    action ViewJobAction = autoRefresh $ viewJob @(jobs)
+    action CreateJobAction = newJob @jobs
+    action DeleteJobAction = deleteJob @jobs
+
+-- | We can't always access the type of our job in order to use type application syntax for 'tableName'.
+-- This is just a convinence function for those cases.
+getTableName :: forall job. (DisplayableJob job) => job -> Text
+getTableName _ = tableName @job
+
+data GenericShowView job = GenericShowView job
+instance (DisplayableJob job) => View (GenericShowView job) where
+    html (GenericShowView job) =
+        let
+            table = getTableName job
+        in [hsx|
+            <br>
+                <h5>Viewing Job {get #id job} in {table}</h5>
+            <br>
+            <table class="table">
+                <tbody>
+                    <tr>
+                        <th>Updated At</th>
+                        <td>{get #updatedAt job}</td>
+                    </tr>
+                    <tr>
+                        <th>Created At</th>
+                        <td>{get #createdAt job}</td>
+                    </tr>
+                    <tr>
+                        <th>Status</th>
+                        <td>{statusToBadge (get #status job)}</td>
+                    </tr>
+                    <tr>
+                        <th>Last Error</th>
+                        <td>{fromMaybe "No error" (get #lastError job)}</td>
+                    </tr>
+                </tbody>
+            </table>
+
+            <div class="d-flex flex-row">
+                <form class="mr-2" action="/jobs/DeleteJob" method="POST">
+                    <input type="hidden" id="tableName" name="tableName" value={table}>
+                    <input type="hidden" id="id" name="id" value={tshow $ get #id job}>
+                    <button type="submit" class="btn btn-danger">Delete</button>
+                </form>
+                <form action="/jobs/CreateJob" method="POST">
+                    <input type="hidden" id="tableName" name="tableName" value={table}>
+                    <button type="submit" class="btn btn-primary">Run again</button>
+                </form>
+            </div>
+        |]
+
+
+data GenericNewJobView job = GenericNewJobView
+instance (DisplayableJob job) => View (GenericNewJobView job) where
+    html _ =
+        let
+            table = tableName @job
+        in [hsx|
+            <br>
+                <h5>New Job: {table}</h5>
+            <br>
+            <form action="/jobs/CreateJob" method="POST">
+                <input type="hidden" id="tableName" name="tableName" value={table}>
+                <button type="submit" class="btn btn-primary">New Job</button>
+            </form>
+        |]
+
+instance HasPath (JobsDashboardController authType jobs) where
+    pathTo _ = error "HasPath not supported for JobsDashboardController."
```

Comment:
Why is this not supported? :) E.g. maybe you want to link to e.g. the `AllJobsAction` from the admin backend

Reply:
Didn't see a way to do this at first, but I found a way! Just had to restructure some things.